### DISCUSSION TITLE
Nextpress: A Wordpress-like CMS based entirely on Next.js?

### DISCUSSION THREAD

**Discussion by darwin403** [vercel/next.js#36033] [Ideas]
I think one of the most attractive features of Next.js is being able to work on both the Frontend + Backend and then seamlessly deploying it anywhere.

WordPress CMS still occupies a major space in publishing but it's heavily outdated in terms of modern technology. One of the reasons WP is still so active is that it's a one-codebase ecosystem which means the developers can modify the frontend and backend seamlessly (something similar to what Next.js did to React.js+Node.js backend). I think investing resources into **a modern alternative for WordPress entirely based on Next.js that can be deployed onto Vercel or a custom server with a click**.

The modern approach has been to use a 3rd party CMS like Sanity, Strapi, etc. but there still is a lot of potential for a combined CMS + Next.js solutions. Ever since the dynamic shift towards JAMStack, Next.js has taken over the competition quick strong. But there is still room to build a nice WP-like ecosystem that would further ease JAMStack CMS's problems and enhance its development. Currently, every JAMStack developer starts from scratch on a Next.js project. But we do we manage to build an ecosystem like WP, we can combine all the developer's efforts of JAMStack projects and further cut down developer times and efficiency on CMS development.

It would just be WordPress with templates, plugins, etc the entire ecosystem is built entirely based on the Next.js framework. The vision is to be able to deploy simple to a complex CMS website onto Vercel.com with no hassle just like WordPress with a single click!

Would Vercel be interested in investing in such an idea? @timneutkens  :D I have a group of buddies who would love to work on this project. Just wanted to get an opinion for the founders of Next.js on what they think!

Any feedback is welcome! 👍

**Comment by dajomu** [comment]
I was thinking about this today and came here to see if anyone else had thought of it.
The one issue is connecting it to some kind of storage.
For my needs I think I could keep the db on memory on the server (not much data/pages that I need), then back up somewhere for when the server restarts etc. That way you wouldn't even have the latency of a db request.
These are some really rough initial thoughts though. It may turn out that this isn't doable for some reason.

**Comment by GabenGar** [comment]
I wouldn't call the process "seamless", especially for deploys outside of Vercel.

**Reply by Solit0123** [reply]
why so? 
we can use some type of Docker to deploy it to our server and ensure compatibility.

**Comment by FullstackWEB-developer** [comment]
Have you heard of cromwellcms?
https://cromwellcms.com
I think they already made it.
I hope ideas like this will push wordpress out soon.

**Reply by Solit0123** [reply]
> Have you heard of cromwellcms? https://cromwellcms.com I think they already made it. I hope ideas like this will push wordpress out soon.

thank you!! 
this is a cool project! 
as someone who build react/nextjs websites for clients this is great! 
However, i wish there was a way to enable regular developers to contribute more easily.

They need to make some video tutorials, their doc is overwhelming

**Reply by darwin403** [reply]
Looks interesting. But still needs a lot more work and I think the vision behind the product is a bit incoherent even though the motivation for it is the same as above. Just my personal opinion.

**Comment by bzybn** [comment]
I also can recommend primocms.org, based on the very nice SvelteKit

**Comment by Manoj-kumar-C** [comment]
I think this will be implemented by vercel team and the community in the futute . 😁

**Reply by Solit0123** [reply]
i would love to volunteer into making this happen!

**Comment by naveen1337** [comment]
Count me in

**Comment by atj4me** [comment]
@darwin403 I think it is best that someone puts a start for this project, as this has been 2 years and still nowhere near perfection. 

An open source solution like Wordpress build on top of Next.JS with admin panel and front end. Developers can extend the CMS using plugins, themes as needed and pluggable into the CMS by non-developers. 

Anyone up for the start of something new?

**Reply by thecmsdojo** [reply]
I am going to work on a prototype

**Reply by Manoj-kumar-C** [reply]
Nice . I like to collab .

**Reply by nok91** [reply]
Would like to collab.

**Reply by neelhabib** [reply]
I am also. Waiting for you code.

**Reply by destinyjibola** [reply]
would love to collab

**Comment by neelhabib** [comment]
Any update on this?

**Comment by joevingracien** [comment]
I think Payload CMS 3.0 is what you're looking for: https://github.com/payloadcms/payload-3.0-demo

Still beta atm, will most likely be stable shortly after both react 19 and next.js 15 are stable.

**Comment by rohit35109** [comment]
May be we should 
@joevingracien i have been thinking of this idea for long but if we have something then lets contribute and get this going.

**Comment by MattatDirectus** [comment]
Howdy! Figured this might be of interest for some of you - 

Worked with Colby Fayock (aka Space Jelly) to create an open source Next.js-based blog platform powered by Directus, based on the WP 2024 theme. 

Here's a link to the repo: [https://github.com/colbyfayock/directus-blog-starter](https://github.com/colbyfayock/directus-blog-starter)

Here's a video overview of Colby building it out (self-hosted via Railway): [https://www.youtube.com/watch?v=S7f41PoKJaw&t=327s](https://www.youtube.com/watch?v=S7f41PoKJaw)

**Reply by Manoj-kumar-C** [reply]
Nice dude

**Comment by davidjeba** [comment]
is this still happening? i want a decent nextjs cms with plugin architecture to use as a base for v0.dev

**Comment by rosariogueli** [comment]
Been thinking about this too. Should support themes and plugins. It'd be great if the admin looked the same as WP, just for consistency and ease of use to capture those 500M millions of existing WP users... Obvs this solution would fit those people that want to move away from php...
